spacy/training/alignment_array.pyx

```diff
@@ -6,28 +6,33 @@ import numpy
 cdef class AlignmentArray:
     """AlignmentArray is similar to Thinc's Ragged with two simplfications:
     indexing returns numpy arrays and this type can only be used for CPU arrays.
-    However, these changes make AlginmentArray more efficient for indexing in a
+    However, these changes make AlignmentArray more efficient for indexing in a
     tight loop."""
 
     __slots__ = []
 
     def __init__(self, alignment: List[List[int]]):
-        self._lengths = None
-        self._starts_ends = numpy.zeros(len(alignment) + 1, dtype="i")
-
         cdef int data_len = 0
         cdef int outer_len
         cdef int idx
+
+        self._starts_ends = numpy.zeros(len(alignment) + 1, dtype='int32')
+        cdef int* starts_ends_ptr = <int*>self._starts_ends.data
```

Comment:
I know that `int` is pretty much fixed, but it always makes me a bit nervous. Maybe `int32_t`? Cython provides [definitions](https://github.com/cython/cython/blob/master/Cython/Includes/libc/stdint.pxd) for these as well.